jib-core/README.md

```diff
@@ -1,3 +1,116 @@
-[![Analytics](https://cloud-tools-for-java-metrics.appspot.com/UA-121724379-2/jib-core)](https://github.com/igrigorik/ga-beacon)
+![experimental](https://img.shields.io/badge/stability-experimental-red.svg)
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/com.google.cloud.tools/jib-core/badge.svg)](https://maven-badges.herokuapp.com/maven-central/com.google.cloud.tools/jib-core)
+[![Gitter version](https://img.shields.io/gitter/room/gitterHQ/gitter.svg)](https://gitter.im/google/jib)
+
+# Jib Core - Java library for building containers
+
+Jib Core is a Java library for building Docker and [OCI](https://github.com/opencontainers/image-spec) container images.
+
+For information about the Jib project, see the [Jib project README](../README.md).\
+For the Maven plugin, see the [jib-maven-plugin project](../jib-maven-plugin).\
+For the Gradle plugin, see the [jib-gradle-plugin project](../jib-gradle-plugin).
+
+## Upcoming features
+
+- Extensions to make building Java and other language-specific containers easier
+- Structured events to react to parts of Jib Core's execution
+
+See [Milestones](https://github.com/GoogleContainerTools/jib/milestones) for planned features. [Get involved with the community](https://github.com/GoogleContainerTools/jib/tree/master#get-involved-with-the-community) for the latest updates.
+
+## Adding Jib Core to your build
+
+Add Jib Core as a dependency using Maven:
+
+```xml
+<dependency>
+  <groupId>com.google.cloud.tools</groupId>
+  <artifactId>jib-core</artifactId>
+  <version>0.1.0</version>
+</dependency>
+```
+
+Add Jib Core as a dependency using Gradle:
+
+```groovy
+dependencies {
+  compile 'com.google.cloud.tools:jib-core:0.1.0'
+}
+```
+
+## Simple example
+
+```java
+Jib.from("busybox")
+   .addLayer(Arrays.asList(Paths.get("helloworld.sh")), AbsoluteUnixPath.get("/")) 
+   .setEntrypoint("/bin/sh", "-c", "chmod +x /helloworld.sh && /helloworld.sh")
+   .containerize(
+       Containerizer.to(RegistryImage.named("gcr.io/my-project/hello-from-jib")
+                                     .addCredential("myusername", "mypassword")));
```

Comment:
I wonder if it's necessary to have this `addCredential()` here?   As a minimal example, its inclusion implies that it's required. Could we remove it and add a note to the explanatory line below that Jib will try to obtain credentials from various sources, but an explicit username/password can also be provided.

Reply:
For Jib Core, `RegistryImage` is not configured automatically with any credential retrievers by default. If the user wanted to use credentials from Docker config automatically, for example, they would need to get the `CredentialRetriever` via `CredentialRetrieverFactory.forImage(...).dockerConfig()`.

Reply:
Is that worth noting?  And/or pointing to `jib-plugins-common`'s `DefaultCredentialRetrievers`?

Reply:
We won't be able to refer to `DefaultCredentialRetrievers` since that is not part of Jib Core, but I can definitely add something about `CredentialRetrieverFactory`.